plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/actions/datasource/DataSourceAutoCommitHandler.java

```diff
@@ -47,6 +51,18 @@
 public class DataSourceAutoCommitHandler extends AbstractDataSourceHandler implements IElementUpdater {
     @Override
     public Object execute(ExecutionEvent event) throws ExecutionException {
+        DBPWorkspace workspace = DBWorkbench.getPlatform().getWorkspace();
+        if (!(workspace.hasRealmPermission(RMConstants.PERMISSION_PROJECT_ADMIN)
+            || workspace.hasRealmPermission(RMConstants.PERMISSION_DATABASE_DEVELOPER))

```

Comment:
And once again, I believe
```suggestion
        if (!workspace.hasRealmPermission(RMConstants.PERMISSION_PROJECT_ADMIN) &&
            !workspace.hasRealmPermission(RMConstants.PERMISSION_DATABASE_DEVELOPER)
```
Is easier to understand